Remove Endpoints

PrettyDG opened the pull request:
- Added endpoint for deleting users and films by ID

PrettyDG committed:
- Added endpoint for deleting users and films by ID

PrettyDG committed:
- Added empty lines in interfaces

h2kb-ya commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -45,6 +45,11 @@ public Film update(@RequestBody @Valid final Film film) {
     }
 
+    @DeleteMapping("/{id}")
+    public void removeFilm(@PathVariable(name = "id") @Positive final Integer filmId) {
```
Думаю, единообразия ради, лучше назвать метод deleteById. Во-первых Film в названии в целом не нужно, так как мы и так работаем в контроллере фильмов, тут по идее ни чего другого удалять не придётся. Ну может быть сериалы, к примеру, когда-нибудь. Во-вторых под капотом removeFilm мы вызываем метод с названием delete. Немного режет глаз, как по мне.

По этой же причине нет смысла, как мне кажется, называть переменную filmId, можно просто id

h2kb-ya commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -37,6 +37,11 @@ public User update(@RequestBody @Valid final User user) {
     }
 
+    @DeleteMapping("/{id}")
+    public void removeUser(@PathVariable(name = "id") @Positive final Integer userId) {
```
Тут аналогично первому комментарию

h2kb-ya commented on src/main/java/ru/yandex/practicum/filmorate/repository/FilmRepository.java:
```diff
@@ -12,6 +12,8 @@ public interface FilmRepository {
 
     void delete(Film film);
 
+    void deleteById(Integer filmId);
```
id

h2kb-ya commented on src/main/java/ru/yandex/practicum/filmorate/repository/FilmRepositoryImpl.java:
```diff
@@ -80,6 +80,14 @@ public void delete(Film film) {
     }
 
+    @Override
+    public void deleteById(Integer filmId) {
```
Метод delete чуть выше, делает тоже самое, думаю, нужно оставить что-то одно, а второе удалить.

h2kb-ya commented on src/main/java/ru/yandex/practicum/filmorate/repository/UserRepository.java:
```diff
@@ -12,6 +12,8 @@ public interface UserRepository {
 
     void delete(User user);
 
+    void deleteById(Integer userId);
```
id

h2kb-ya commented on src/main/java/ru/yandex/practicum/filmorate/repository/UserRepositoryImpl.java:
```diff
@@ -68,6 +68,14 @@ public void delete(User user) {
     }
 
+    @Override
+    public void deleteById(Integer userId) {
```
Тут тоже есть метод delete, с аналогичным функционалом

h2kb-ya commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmServiceImpl.java:
```diff
@@ -71,6 +71,13 @@ public Film update(final Film film) {
     }
 
+    @Override
+    public void deleteById(Integer filmId) {
```
А если смотреть тесты постмана, у нас нет контракта, что если при удалении сущности нет, то вернуть NotFound?

h2kb-ya commented on src/test/java/ru/yandex/practicum/filmorate/repository/FilmRepositoryImplIntegrationTest.java:
```diff
@@ -141,6 +141,17 @@ public void delete_filmExists_filmDeleted() {
     }
 
+    @Test
+    public void deleteFilm_byFilmId() {
```
Есть практика называть тесты тестируемыйМетод_условие(например, itemExists)_ожидаемый результат
deleteById_filmExists_filmDeleted()

h2kb-ya commented on src/test/java/ru/yandex/practicum/filmorate/repository/FilmRepositoryImplIntegrationTest.java:
```diff
@@ -141,6 +141,17 @@ public void delete_filmExists_filmDeleted() {
+    public void deleteFilm_byFilmId() {
+        Integer filmId = 1;
+        Optional<Film> existedFilm = filmRepositoryImpl.findById(filmId);
+        assertThat(existedFilm).isNotEmpty();
```
Тест delete_filmExists_filmDeleted проверяет тоже самое

h2kb-ya commented on src/test/java/ru/yandex/practicum/filmorate/repository/UserRepositoryImplIntegrationTest.java:
```diff
@@ -106,6 +106,17 @@ public void delete_userExists_userDeleted() {
     }
 
+    @Test
+    public void deleteUser_byUserId() {
```
Тест delete_userExists_userDeleted() делает тоже самое

PrettyDG committed:
Merge branch 'develop' into add-remove-endpoint

PrettyDG committed:
- Refactor variable names
- Deleted unnecessary methods and tests
- Add throwing NotFound films and users

h2kb-ya commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -13,6 +13,8 @@ public interface FilmService {
 
     Film update(Film film);
 
+    void deleteById(Integer filmId);
```
id

h2kb-ya commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmServiceImpl.java:
```diff
@@ -71,6 +71,15 @@ public Film update(final Film film) {
 
+    @Override
+    public void deleteById(Integer id) {
+        if (filmRepository.findById(id).isPresent()) {
```
А если вот так, без if?
`filmRepository.findById(id).ifPresentOrElse(
                film -> filmRepository.deleteById(film.getId()),
                () -> {
                    throw new NotFoundException("Фильм c id " + id + " не найден.");
                });`

h2kb-ya commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -13,6 +13,8 @@ public interface UserService {
 
     User update(User user);
 
+    void deleteById(Integer userId);
```
id

h2kb-ya commented on src/test/java/ru/yandex/practicum/filmorate/repository/FilmRepositoryImplIntegrationTest.java:
```diff
@@ -136,7 +136,7 @@ public void delete_filmExists_filmDeleted() {
         assertThat(existedFilm).isNotEmpty();
 
-        filmRepositoryImpl.delete(getTestFilm1());
+        filmRepositoryImpl.deleteById(getTestFilm1().getId());
```
Давай в имени теста тоже поправим на deleteById?

h2kb-ya commented on src/test/java/ru/yandex/practicum/filmorate/repository/UserRepositoryImplIntegrationTest.java:
```diff
@@ -101,7 +101,7 @@ public void delete_userExists_userDeleted() {
         assertThat(existedUser).isNotEmpty();
 
-        userRepositoryImpl.delete(getTestUser1());
+        userRepositoryImpl.deleteById(getTestUser1().getId());
```
Давай в имени теста тоже поправим на deleteById?

Commit:
Merge branch 'develop' into add-remove-endpoint

PrettyDG committed:
- Refactor variable names

h2kb-ya reviewed (approved):
Nice job @PrettyDG
